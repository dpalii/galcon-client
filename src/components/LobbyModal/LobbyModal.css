.backdrop {
    position: fixed;
    width: 100vw;
    height: 100vh;
    background-color: rgba(0, 0, 0, 0.5);
    z-index: 1;
}

.modal {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    background-color: var(--bg-primary);
    border-radius: 8px;
    padding: 20px;
}

.heading {
    margin: 0;
    padding-bottom: 20px;
    font-size: 1.2rem;
    font-weight: bold;
}

.close-btn {
    color: var(--text-default);
    border: none;
    border-radius: 4px;
    padding: 10px 15px;
    background-color: var(--bg-default);
    font-weight: bold;
    transition: background-color 0.3s;
    cursor: pointer;
}

.close-btn:hover {
    background-color: var(--bg-default-dark);
}

.close-btn:active {
    background-color: var(--bg-default-darker);
}

.close-btn:focus {
    outline: none;
}
