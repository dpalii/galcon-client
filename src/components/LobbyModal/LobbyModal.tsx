import './LobbyModal.css';

export interface LobbyModalProps {
    players: string[];
    close: () => void;
}

export function LobbyModal({players, close}: LobbyModalProps) {

    return (
        <div className='backdrop' onClick={() => close()}>
            <div className='modal' onClick={(e) => e.stopPropagation()}>
                <h2 className='heading'>PLAYERS:</h2>
                <button className='close-btn' onClick={() => close()}>CLOSE</button>
            </div>
        </div>
    )
}