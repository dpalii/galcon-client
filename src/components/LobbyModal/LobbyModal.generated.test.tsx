/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/no-unnecessary-act */
import { render } from "react-dom";
import { fireEvent} from '@testing-library/react'

import { unmountComponentAtNode } from "react-dom";
import { act } from 'react-dom/test-utils';
import { LobbyModal } from "../../../src/components/LobbyModal/LobbyModal";

let container: any = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('<LobbyModal>', () => {
  it('should render component', () => {
    act(() => {
      render(<LobbyModal  
        players={[{
          id: 'string',
          name: 'string',
          color: '#000000',
          isHost: true
      }, {
        id: 'string2',
        name: 'string2',
        color: '#ffffff',
        isHost: false
    }]}  
        close={() => {}} 
      />, container);
    });
    container.querySelector('.backdrop').dispatchEvent(new MouseEvent("click", { bubbles: true }));
    container.querySelector('.btn').dispatchEvent(new MouseEvent("click", { bubbles: true }));
    expect(container).toMatchSnapshot();
  })
});

