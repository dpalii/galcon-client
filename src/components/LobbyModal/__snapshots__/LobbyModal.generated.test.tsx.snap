// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LobbyModal> should render component 1`] = `
<div>
  <div
    class="backdrop"
  >
    <div
      class="modal"
    >
      <h2
        class="heading"
      >
        PLAYERS:
      </h2>
      <div
        class="player-entry"
      >
        <span
          class="player-color"
          style="background-color: rgb(0, 0, 0);"
        />
        <b>
          string
           
          (host)
        </b>
      </div>
      <div
        class="player-entry"
      >
        <span
          class="player-color"
          style="background-color: rgb(255, 255, 255);"
        />
        <b>
          string2
           
        </b>
      </div>
      <button
        class="btn w-100"
      >
        CLOSE
      </button>
    </div>
  </div>
</div>
`;
